crates/ruff_linter/src/rules/flake8_pyi/rules/redundant_none_literal.rs

```diff
@@ -167,3 +181,82 @@ fn create_fix_edit(semantic: &SemanticModel, literal_expr: &Expr) -> Option<Edit
 
     is_fixable.then(|| Edit::range_replacement("None".to_string(), literal_expr.range()))
 }
+
+
+fn create_fix_edit_2(checker: &mut Checker, literal_expr: &Expr, literal_elements: &[&Expr]) -> Option<Edit> {
+    let enclosing_pep604_union = checker.semantic()
+        .current_expressions()
+        .skip(1)
+        .take_while(|expr| {
+            matches!(
+                expr,
+                Expr::BinOp(ExprBinOp {
+                    op: Operator::BitOr,
+                    ..
+                })
+            )
+        })
+        .last();
+
+    let mut is_fixable = true;
+    if let Some(enclosing_pep604_union) = enclosing_pep604_union {
+        traverse_union(
+            &mut |expr, _| {
+                if matches!(expr, Expr::NoneLiteral(_)) {
+                    is_fixable = false;
+                }
+                if expr != literal_expr {
+                    if let Expr::Subscript(ExprSubscript { value, slice, .. }) = expr {
+                        if checker.semantic().match_typing_expr(value, "Literal")
+                            && matches!(**slice, Expr::NoneLiteral(_))
+                        {
+                            is_fixable = false;
+                        }
+                    }
+                }
+            },
+            checker.semantic(),
+            enclosing_pep604_union,
+        );
+    }
+
+    let bin_or = Expr::BinOp(
+        ExprBinOp {
+            range: TextRange::default(),
+            left: Box::new(Expr::Subscript(ast::ExprSubscript {
+                value: Box::new(ruff_python_ast::Expr::Name(ast::ExprName{
+                    range: TextRange::default(),
+                    id: Name::new("Literal"),
```

Comment:
this will only work if the user has `from typing import Literal` somewhere in the file. If the user has `import typing` somewhere in the file instead of `from typing import Literal`, the fix here will break the user's code, which won't be good! Instead of using an `ExprName` that we're constructing ourselves as the subscript value, we should just use the value of the original subscript expression

Reply:
Oh yes, I totally missed this. Thanks!